smithy-typescript-ssdk-libs/server-node/src/index.spec.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { mkdtemp } from "fs/promises";
+import { createServer, IncomingMessage, request, RequestOptions, Server, ServerResponse } from "http";
+import * as os from "os";
+import * as path from "path";
+import { Readable } from "stream";
+
+import { convertRequest } from "./node";
+
+let socketPath: string;
+let promiseResolve: ([req, res]: [IncomingMessage, ServerResponse]) => void;
+
+let server: Server;
+beforeAll(async () => {
+  server = createServer(function (req, res) {
+    promiseResolve([req, res]);
+    resToEnd = res;
+  });
+  // Create a temporary named pipe where to run the server and obtain a request
+  socketPath = path.join(await mkdtemp(path.join(os.tmpdir(), "named-pipe-for-test-")), "server");
```

Comment:
Why listen to a socket rather than a port? port 0 would bind to any free port.

Reply:
I'm not a huge fan of tests that depend on the TCP stack in general (probably since I have had clash with already used ports many times). It's also true that I can bind to the socket (as you describe) and it's probably going to be still quite fast. Do you think I'm losing performance or how realistic is the scenario under test?